### DISCUSSION TITLE
Does domain decomposition work with structured grids/meshs?

### DISCUSSION THREAD

**Discussion by DeniseWorthen** [NOAA-EMC/WW3#944] [General]
There are several locations where the conditional is used:

``IF (.NOT. LPDLIB .or. (GTYPE.ne.UNGTYPE))``

This evaluates as true for the following cases:
1) Card-deck decomposition and any grid type (structured or unstructured)
2) Domain-decomposition and not unstructured

Is the second case possible? If domain decomposition can only run with unstructured domains, then a simple ``if (.not. lpdlib)`` would be sufficient and the double-negative construction could be simplified.

**Comment by JessicaMeixner-NOAA** [comment]
Domain decomposition cannot be used with structured grids.

**Reply by DeniseWorthen** [reply]
That is what I believe also, but the code allows for that possibility. I think the current construct is not helpful---if case 2 cannot exist, then a simple check on domain decomposition library is sufficient.

**Comment by aliabdolali** [comment]
We never checked the meshcap with wetting and drying. This is a place we need to investigate after all urgent developments for the sake of GFSv17 are over.

**Reply by DeniseWorthen** [reply]
I'm not clear on what wetting and drying has to do w/ domain decomp and grid type?

**Reply by aliabdolali** [reply]
> I'm not clear on what wetting and drying has to do w/ domain decomp and grid type?

what we did so far was w/o wetting and drying, and all meshes I created were with mom6 mask, which is sufficient for GFS. I got confirmation from @AvichalMehra-NOAA that on global scale, we do not do wetting and drying. So we are good for now.
But, if wetting and drying is activated, the masked areas might change, and affect interpolation. I recall you and I discussed it. I am more than happy to discuss them further

**Comment by aliabdolali** [comment]
@DeniseWorthen PDLIB is self-contained, if it is selected, it is for domain decomposition. No extra check needed.

**Reply by DeniseWorthen** [reply]
@aliabdolali @JessicaMeixner-NOAA Thanks for confirming.
